src/components/Modals/NewCourse/AddCourseFromSemesterModalNewFlow.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+  <TeleportModal
+    title="Add Course"
+    content-class="content-add-new-course-from-semester"
+    :leftButtonText="leftButtonText"
+    right-button-text="Add"
+    :rightButtonIsDisabled="selectedCourse == null"
```

Comment:
This checks both null and undefined. It's generally a better practice, since it's usually a bad idea to distinguish your program behavior based on `null` vs `undefined`, so it's better to have a check to handle them all instead of thinking about what might be the precise case.

Read more: https://stackoverflow.com/questions/23101407/javascript-best-practice-to-check-only-null-values